lib/Dialect/FIRRTL/FIRRTLOps.cpp

```diff
@@ -105,25 +114,25 @@ static MemOp::PortKind getMemPortKindFromType(FIRRTLType type) {
   // Get the kind of port based on the fields of the Bundle.
   for (auto elem : portType.getElements()) {
     fields |= llvm::StringSwitch<unsigned>(elem.name.getValue())
-                  .Case("addr", 1)
-                  .Case("en", 2)
-                  .Case("clk", 4)
-                  .Case("data", 8)
-                  .Case("mask", 16)
-                  .Case("rdata", 32)
-                  .Case("wdata", 64)
-                  .Case("wmask", 128)
-                  .Case("wmode", 256)
+                  .Case("addr", addr)
+                  .Case("en", en)
+                  .Case("clk", clk)
+                  .Case("data", data)
+                  .Case("mask", mask)
+                  .Case("rdata", rdata)
+                  .Case("wdata", wdata)
+                  .Case("wmask", wmask)
+                  .Case("wmode", wmode)
                   .Default(512);
   }
   // addr, en, clk, data
-  if (fields == 15)
+  if (fields == (addr | en | clk | data))
     return MemOp::PortKind::Read;
   // addr, en, clk, data, mask
-  if (fields == 31)
+  if (fields == (addr | en | clk | data | mask))
     return MemOp::PortKind::Write;
   // addr, en, clk, wdata, wmask, rdata, wmode
-  if (fields == 487)
+  if (fields == (addr | en | clk | wdata | wmask | rdata | wmode))
```

Comment:
Consider sorting these in (ascending) order so it's easier to see which are missing-- here it looks like this is everything `wmode` and down excluding `data` and `mask`.  Maybe this makes sense as-is for other reasons, just a thought.

Reply:
Actually, the order is a convention for memory ports, it is followed while emitting the ports. So, lets stick with it.